packages/amplify-e2e-core/src/categories/lambda-layer.ts

```diff
@@ -2,28 +2,25 @@ import fs from 'fs-extra';
 import _ from 'lodash';
 import path from 'path';
 import { nspawn as spawn, ExecutionContext, getCLIPath, KEY_DOWN_ARROW } from '../../src';
+import { getLayerVersion, listVersions } from '../utils/sdk-calls';
 import { multiSelect } from '../utils/selectors';
 
 type LayerRuntimes = 'dotnetcore3.1' | 'go1.x' | 'java' | 'nodejs' | 'python';
 
 const layerRuntimeChoices = ['NodeJS', 'Python'];
 const permissionChoices = ['Specific AWS accounts', 'Specific AWS organization', 'Public (Anyone on AWS can use this layer)'];
 
-export function validateLayerDir(projRoot: string, layerName: string, layerExists: boolean, runtimes: LayerRuntimes[]): boolean {
+export function validateLayerDir(projRoot: string, layerName: string, runtimes: LayerRuntimes[]): boolean {
   let layerDir = path.join(projRoot, 'amplify', 'backend', 'function', layerName);
-  if (layerExists) {
-    let validDir = fs.pathExistsSync(path.join(layerDir, 'opt'));
-    if (runtimes && runtimes.length) {
-      for (let runtime of runtimes) {
-        if (!fs.pathExistsSync(path.join(layerDir, getLayerRuntimeInfo(runtime).path))) {
-          return false;
-        }
+  let validDir = fs.pathExistsSync(path.join(layerDir, 'opt'));
+  if (runtimes && runtimes.length) {
+    for (let runtime of runtimes) {
```

Comment:
nit - could this be done with a foreach loop?

Reply:
Yes, but since I'm returning from the for loop, that's gonna require some slightly different logic. Unless there's a compelling reason I'm missing, I'm going to leave as is.